### ISSUE TITLE
Separating features/support/world.js files

### ISSUE THREAD

**Issue by Zsuark** [AgileVentures/AsyncVoter#62]
To avoid git conflicts, and to keep everything clear, cucumber step definitions should have differing `features/support/world.js` files.

E.g. for `features/cast_vote.feature` there should be a `features/support/cast_vote_world.js`, and for `features/start_vote.feature` there should be a `features/support/start_vote_world.js` file.

**Comment by tansaku** [comment]
any DRYing of replication is a potential dependency - I don't see a strong need to do this yet - I would probably let myself get bitten by the dependency before working to extract it, particularly since we have yet to get to our MVP ...

**Comment by arreche** [comment]
I think it would be convenient understanding and cleaning up that  [world file](https://github.com/AgileVentures/AsyncVoter/blob/master/features/support/world.js) before to write  any cucumber test.

- https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/world.md
- https://www.custardbelly.com/blog/blog-posts/2014/01/22/cucumberjs-world/index.html

**Comment by tansaku** [comment]
notes from slack

```

Sam Joseph	[4:03 PM]  
@arreche @zsuark I'd strongly advise at least starting the vote

[4:04]  
nothing stops you spiking anything at any time - you don't have to wait for the vote to finish - but if there's no other vote in progress, there's nothing to lose by starting the vote, and you increase the awareness across the team of the feature or issue you are interested in

Raphael Krausz	[4:04 PM]  
We agree - but no-one is around, except us two

[4:04]  
(or were)

Sam Joseph	[4:05 PM]  
I don't think you know that for sure - if you start a vote with at-channel you increase the chances of people seeing, and there can be good input and discussion going on in parallel with your spiking :slightly_smiling_face:

Alex Kiss	[4:09 PM]  
I am actually always here

[4:09]  
just caught in some other time consuming actions

Raphael Krausz	[4:09 PM]  
ahh :smile:

[4:09]  
One min

arreche nash	[4:14 PM]  
@channel :speaking_head_in_silhouette: New [ASYNC VOTE] now on: https://github.com/AgileVentures/AsyncVoter/issues/62 Separating world.js files. Please DM me with: 1(Simple), 2 (Medium) or 3(Hard) - Discussion in ticket or here as you prefer. :slightly_smiling_face:

Sam Joseph	[4:18 PM]  
thanks @arreche commented in ticket - feels like low priority to me - key thing is to get that feature that @chaiwa is working on merged in - but I guess we need to give him some time to sync up - I'd say documenting the API is higher priority than having multiple world files - just my opinion :slightly_smiling_face: everyone free to work on their preferred areas of the app :slightly_smiling_face:

Raphael Krausz	[4:20 PM]  
I think this is a good time - we only have a couple of hours, and is do-able and makes sense moving forward.

[4:20]  
Later, it will be harder and take longer.  Doing it now makes it simpler moving forward.

Sam Joseph	[4:21 PM]  
yes, but if we then end up replicating lots of similar code across multiple files it might make it more difficult to maintain - I'd say we learn that as we go

Raphael Krausz	[4:21 PM]  
If we keep adding to a global file, there will be many more parts to separate out.

Sam Joseph	[4:21 PM]  
I'd say easier to do later - but we can agree to disagree :slightly_smiling_face:

Raphael Krausz	[4:21 PM]  
We can put common functionality in a common file,

[4:22]  
... separate sections out for cleaner code...

[4:22]  
It's already starting to get difficult to read, and @arreche has mentioned this multiple times.

[4:22]  
(At least to me when pairing)
```
